### PR TITLE
[Profiling] Use default task cancellation check

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we remove our custom implementation of whether a task has been cancelled and instead use the standard implementation that is already provided by the task API.

### CODE REVIEW

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java [hunk 1]
```diff
@@ -155,22 +154,6 @@ protected void doExecute(Task submitTask, GetStackTracesRequest request, ActionL
...
-     * @return <code>true</code> iff the task has been cancelled. Callers must terminate as early as possible.
-     */
-    private boolean mayNotifyOfCancellation(Task task, ActionListener<GetStackTracesResponse> listener) {
-        if (task instanceof CancellableTask && ((CancellableTask) task).isCancelled()) {
-            log.info("{} got cancelled.", task);
-            listener.onFailure(new TaskCancelledException("get stacktraces task cancelled"));
-            return true;
-        } else {
-            return false;
-        }
```
Comment (jbcrail) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java:171]: Does the default implementation record whether the task was cancelled? Just curious if this is something that we should care about.

**Review by jbcrail — APPROVED** [review]
Nice, looks good :+1:

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java [hunk 2]
```diff
@@ -447,7 +430,7 @@ private void retrieveStackTraces(
         GetStackTracesResponseBuilder responseBuilder,
         ActionListener<GetStackTracesResponse> submitListener
     ) {
-        if (mayNotifyOfCancellation(submitTask, submitListener)) {
+        if (submitTask instanceof CancellableTask c && c.notifyIfCancelled(submitListener)) {
```
Comment (DaveCTurner) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java:433]: I think `submitTask` is _always_ a `CancellableTask`, it'd be best to change the argument type to reflect this (and cast it to `CancellableTask` in `doExecute` instead).

Comment (danielmitterdorfer) [reply x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java:433]: Thanks! I've pushed https://github.com/elastic/elasticsearch/pull/107139.

**Pull request merged**